tests/test_default_qubit.py

```diff
@@ -544,6 +544,94 @@ def test_var_hermitian_rotated_state(self):
                     *np.cos(phi)*(np.sin(theta)-np.cos(theta))+35*np.cos(2*phi)+39)
         self.assertAlmostEqual(var, expected, delta=self.tol)
 
+    def test_sample_dimensions(self):
+        """Tests if the samples returned by sample have 
+        the correct dimensions
+        """
+        self.logTestName()
+        self.dev.reset()
+
+        self.dev.apply('RX', wires=[0], par=[1.5708])
+        self.dev.apply('RX', wires=[1], par=[1.5708])
+
+        s1 = self.dev.sample('PauliZ', [0], [], 10)
+        self.assertAllEqual(s1.shape, (10,))
+
+        s2 = self.dev.sample('PauliZ', [1], [], 12)
+        self.assertAllEqual(s2.shape, (12,))
+
+        s3 = self.dev.sample('CZ', [0,1], [], 17)
+        self.assertAllEqual(s3.shape, (17,))
+
+    def test_sample_values(self):
+        """Tests if the samples returned by sample have
+        the correct values
+        """
+        self.logTestName()
+        self.dev.reset()
+
+        self.dev.apply('RX', wires=[0], par=[1.5708])
+
+        s1 = self.dev.sample('PauliZ', [0], [], 10)        
+
+        # s1 should only contain 1 and -1, which is guaranteed if
+        # they square to 1
+        self.assertAllAlmostEqual(s1**2, 1, delta=self.tol)
+
+    def test_sample_mean_and_variance(self):
+        """Tests if the sampling works as expected by checking mean
+        and variance
+        """
+        self.logTestName()
+        self.dev.reset()
+
+        self.dev.apply('RX', wires=[0], par=[1.570796])
+
+        s1 = self.dev.sample('PauliZ', [0], [], 2000)
+
+        # We use large deltas here because sampling is stochastic
+        # this ensures that the test only fails if something in
+        # the sampling itself goes awry
```

Comment:
We've kind of decided that statistical tests don't belong in the same place as unit tests (though I'm sure there's still some vestigial traces of this thinking throughout our codebase). This follows the example of `numpy` which does not have any stochastic tests for its `np.random` submodule. 

I'd prefer to just remove this test. You're already testing the important parts, namely the shape and type of returns, and special cases where the result is non-deterministic, so that's enough for me :+1:

Reply:
Good!